pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -1304,6 +1357,305 @@ def plot_new_spend_contributions(
     def _validate_data(self, X, y=None):
         return X
 
+    def _channel_map_scales(self) -> dict:
+        return dict(
+            zip(
+                self.channel_columns,
+                self.channel_transformer["scaler"].scale_,
+                strict=False,
+            )
+        )
+
+    def format_recovered_transformation_parameters(
+        self, quantile: float = 0.5
+    ) -> dict[str, dict[str, dict[str, float]]]:
+        """
+        Format the recovered transformation parameters for each channel.
+
+        This function retrieves the quantile of the parameters for each channel and formats them into a dictionary
+        containing the channel name, the saturation parameters, and the adstock parameters.
+
+        Parameters
+        ----------
+        quantile : float, optional
+            The quantile to retrieve from the posterior distribution of the parameters. Default is 0.5.
+
+        Returns
+        -------
+        dict
+            A dictionary containing the channel names as keys and the corresponding saturation and adstock parameters
+            as values.
+
+        Example
+        -------
+        >>> self.format_recovered_transformation_parameters(quantile=.5)
+        >>> Output:
+        {
+            'x1': {
+                'saturation_params': {
+                    'lam': 2.4761893929757077,
+                    'beta': 0.360226791880304
+                },
+            'adstock_params': {
+                'alpha': 0.39910387900504796
+                }
+            },
+            'x2': {
+                'saturation_params': {
+                    'lam': 2.6485978655163436,
+                    'beta': 0.2399381337197204
+                },
+            'adstock_params': {
+                'alpha': 0.18859423763437405
+                }
+            }
+        }
+        """
+        # Retrieve channel names
+        channels = self.fit_result.channel.values
+
+        # Initialize the dictionary to store channel information
+        channels_info = {}
+
+        # Define the parameter groups for consolidation
+        param_groups = {
+            "saturation_params": self.saturation.model_config.keys(),
+            "adstock_params": self.adstock.model_config.keys(),
+        }
+
+        # Iterate through each channel to fetch and store parameters
+        for channel in channels:
+            channel_info = {}
+
+            # Process each group of parameters (saturation and adstock)
+            for group_name, params in param_groups.items():
+                # Build dictionary for the current group of parameters
+                param_dict = {
+                    param.replace(group_name[:-7] + "_", ""): self.fit_result[param]
```

Comment:
Where is this `-7` coming from? Can we extract this step as a small function just doing `param.replace(group_name[:-7] + "_", "")` and add some unit tests to make sure we document the expected behaviour? These replacements are often sources of bugs for developers who did not write the code ;)

Reply:
All parameter are call `ParameterName_params` the -7 is to delete the `. . ._params` and hold only **saturation** or **adstock**.